addons/hr_work_entry_contract/static/src/components/work_entry_source_field/work_entry_source_field.js

```diff
@@ -0,0 +1,36 @@
+/** @odoo-module **/
+
+import { _t } from "@web/core/l10n/translation";
+import { evaluateExpr } from "@web/core/py_js/py";
+import { registry } from "@web/core/registry";
+import { RadioField, radioField } from "@web/views/fields/radio/radio_field";
+
+export class WorkEntrySourceField extends RadioField {
+    static template = "hr_work_entry_contract.WorkEntrySourceField";
+    static props = {
+        ...RadioField.props,
+    };
+
+    setup() {
+        super.setup()
+        this.is_fully_flexible = !this.props.record.data.resource_calendar_id;
+        this.initialSelection = this.props.record.data[this.props.name];
+        this.options = this.props.record.fields[this.props.name].selection;
+    };
+
+    // prevent User from choosing `calendar` when resource_calendar is full flexible
+    get items() {
+        if (this.is_fully_flexible) {
+            return this.options.filter(([value, label]) => value !== 'calendar');
+        }
+        return this.options;
+    }
+    // TODO: how to make the above rerender when resource_calendar_id changes?
```

Comment:
I think when the resource_calendar_id will be changed, the form renderer will be rendered and so this component will also be re-rendered as well. Thus, with the getter, we will be able to read the actual value of resource_calendar_id field. 🙂 

To check. 🙂